V2 ImportStateVerify Reporting Incorrect Resource Type Attributes

### SDK version

```
v2.0.1
```

### Relevant provider source code

```go
func TestAccAWSVolumeAttachment_skipDestroy(t *testing.T) {
	var i ec2.Instance
	var v ec2.Volume
	resourceName := "aws_volume_attachment.test"
	rName := acctest.RandomWithPrefix("tf-acc-test")

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckVolumeAttachmentDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccVolumeAttachmentConfigSkipDestroy(rName),
				Check: resource.ComposeTestCheckFunc(
					resource.TestCheckResourceAttr(resourceName, "device_name", "/dev/sdh"),
					testAccCheckInstanceExists("aws_instance.test", &i),
					testAccCheckVolumeExists("aws_ebs_volume.test", &v),
					testAccCheckVolumeAttachmentExists(resourceName, &i, &v),
				),
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateIdFunc: testAccAWSVolumeAttachmentImportStateIDFunc(resourceName),
				ImportStateVerify: true,
				ImportStateVerifyIgnore: []string{
					"skip_destroy", // attribute only used on resource deletion
				},
			},
		},
	})
}

func testAccAWSVolumeAttachmentImportStateIDFunc(resourceName string) resource.ImportStateIdFunc {
	return func(s *terraform.State) (string, error) {
		rs, ok := s.RootModule().Resources[resourceName]
		if !ok {
			return "", fmt.Errorf("Not found: %s", resourceName)
		}
		return fmt.Sprintf("%s:%s:%s", rs.Primary.Attributes["device_name"], rs.Primary.Attributes["volume_id"], rs.Primary.Attributes["instance_id"]), nil
	}
}
```

### Terraform Configuration Files

Simplified from above code. Substitute format string verb with random name. No clue why the `aws_ebs_volume` data source in the configuration (seems extraneous), but certainly seems wrong that is what resource state is being used for `ImportStateVerify`. 🤔 

```terraform
data "aws_ami" "amzn-ami-minimal-hvm-ebs" {
  most_recent = true
  owners      = ["amazon"]

  filter {
    name   = "name"
    values = ["amzn-ami-minimal-hvm-*"]
  }

  filter {
    name   = "root-device-type"
    values = ["ebs"]
  }
}

data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}

data "aws_ec2_instance_type_offering" "available" {
  filter {
    name   = "instance-type"
    values = ["t3.micro", "t2.micro"]
  }

  filter {
    name   = "location"
    values = [data.aws_availability_zones.available.names[0]]
  }

  location_type            = "availability-zone"
  preferred_instance_types = ["t3.micro", "t2.micro"]
}

resource "aws_instance" "test" {
  ami               = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
  availability_zone = data.aws_availability_zones.available.names[0]
  instance_type     = data.aws_ec2_instance_type_offering.available.instance_type

  tags = {
    Name = "tf-acc-test-12345678"
  }
}

resource "aws_ebs_volume" "test" {
  availability_zone = data.aws_availability_zones.available.names[0]
  size              = 1

  tags = {
    Name = "tf-acc-test-12345678"
  }
}

data "aws_ebs_volume" "test" {
  filter {
    name   = "size"
    values = [aws_ebs_volume.test.size]
  }
  filter {
    name   = "availability-zone"
    values = [aws_ebs_volume.test.availability_zone]
  }
  filter {
    name   = "tag:Name"
    values = ["tf-acc-test-12345678"]
  }
}

resource "aws_volume_attachment" "test" {
  device_name  = "/dev/sdh"
  volume_id    = data.aws_ebs_volume.test.id
  instance_id  = aws_instance.test.id
  skip_destroy = true
}
```

### Debug Output

Please ask if you need this.

### Expected Behavior

`ImportStateVerify` testing passing as it did in SDK v1 or any expected behavior changes documented in the SDK v2 upgrade guide. 

### Actual Behavior

```
=== CONT  TestAccAWSVolumeAttachment_skipDestroy
TestAccAWSVolumeAttachment_skipDestroy: resource_aws_volume_attachment_test.go:51: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
(map[string]string) (len=2) {
(string) (len=1) "%": (string) (len=2) "12",
(string) (len=4) "type": (string) (len=3) "gp2"
}
(map[string]string) (len=17) {
(string) (len=1) "%": (string) (len=2) "15",
(string) (len=8) "filter.#": (string) (len=1) "3",
(string) (len=10) "filter.0.%": (string) (len=1) "2",
(string) (len=13) "filter.0.name": (string) (len=17) "availability-zone",
(string) (len=17) "filter.0.values.#": (string) (len=1) "1",
(string) (len=17) "filter.0.values.0": (string) (len=10) "us-west-2a",
(string) (len=10) "filter.1.%": (string) (len=1) "2",
(string) (len=13) "filter.1.name": (string) (len=4) "size",
(string) (len=17) "filter.1.values.#": (string) (len=1) "1",
(string) (len=17) "filter.1.values.0": (string) (len=1) "1",
(string) (len=10) "filter.2.%": (string) (len=1) "2",
(string) (len=13) "filter.2.name": (string) (len=8) "tag:Name",
(string) (len=17) "filter.2.values.#": (string) (len=1) "1",
(string) (len=17) "filter.2.values.0": (string) (len=31) "tf-acc-test-7089869605642141730",
(string) (len=11) "most_recent": (string) (len=5) "false",
(string) (len=9) "volume_id": (string) (len=21) "vol-03ab759cf4a102619",
(string) (len=11) "volume_type": (string) (len=3) "gp2"
}
--- FAIL: TestAccAWSVolumeAttachment_skipDestroy (101.17s)
```

Running it another time, it flips 🙃 

```
=== CONT  TestAccAWSVolumeAttachment_skipDestroy
    TestAccAWSVolumeAttachment_skipDestroy: resource_aws_volume_attachment_test.go:51: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.

        (map[string]string) (len=17) {
         (string) (len=1) "%": (string) (len=2) "15",
         (string) (len=8) "filter.#": (string) (len=1) "3",
         (string) (len=10) "filter.0.%": (string) (len=1) "2",
         (string) (len=13) "filter.0.name": (string) (len=17) "availability-zone",
         (string) (len=17) "filter.0.values.#": (string) (len=1) "1",
         (string) (len=17) "filter.0.values.0": (string) (len=10) "us-west-2a",
         (string) (len=10) "filter.1.%": (string) (len=1) "2",
         (string) (len=13) "filter.1.name": (string) (len=4) "size",
         (string) (len=17) "filter.1.values.#": (string) (len=1) "1",
         (string) (len=17) "filter.1.values.0": (string) (len=1) "1",
         (string) (len=10) "filter.2.%": (string) (len=1) "2",
         (string) (len=13) "filter.2.name": (string) (len=8) "tag:Name",
         (string) (len=17) "filter.2.values.#": (string) (len=1) "1",
         (string) (len=17) "filter.2.values.0": (string) (len=31) "tf-acc-test-3995183175188423361",
         (string) (len=11) "most_recent": (string) (len=5) "false",
         (string) (len=9) "volume_id": (string) (len=21) "vol-046c8a254b85d50a5",
         (string) (len=11) "volume_type": (string) (len=3) "gp2"
        }


        (map[string]string) (len=2) {
         (string) (len=1) "%": (string) (len=2) "12",
         (string) (len=4) "type": (string) (len=3) "gp2"
        }
--- FAIL: TestAccAWSVolumeAttachment_skipDestroy (152.14s)
```

### Steps to Reproduce

1. `TF_ACC=1 go test ./aws -v -count 1 -timeout 120m -parallel 20 -run='TestAccAWSVolumeAttachment_skipDestroy'`

### References

Potentially related to:

- #521 (although this report is only with a single provider instance)
- #531 (if `ImportStateCheck` and `ImportStateVerify` use similar logic)

--- a/helper/resource/testing_new_import_state.go
+++ b/helper/resource/testing_new_import_state.go
@@ -110,7 +110,15 @@ func testStepNewImportState(t testing.T, c TestCase, helper *tftest.Helper, wd *
 		for _, r := range new {
 			// Find the existing resource
 			var oldR *terraform.ResourceState
-			for _, r2 := range old {
+			for r2Key, r2 := range old {
+				// Ensure that we do not match against data sources as they
+				// cannot be imported and are not what we want to verify.
+				// Mode is not present in ResourceState so we use the
+				// stringified ResourceStateKey for comparison.
+				if strings.HasPrefix(r2Key, "data.") {
+					continue
+				}
+
 				if r2.Primary != nil && r2.Primary.ID == r.Primary.ID && r2.Type == r.Type && r2.Provider == r.Provider {
 					oldR = r2
 					break

